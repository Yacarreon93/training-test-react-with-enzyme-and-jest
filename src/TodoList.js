import React from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";

import { addTodo, removeTodo } from "./store";

export class TodoList extends React.Component {
  state = {
    todo: ""
  };

  addTodo = () => {
    if (this.state.todo) {
      this.props.addTodo({ text: this.state.todo });
      this.setState({ todo: "" });
    }
  };

  removeTodo = id => {
    this.props.removeTodo(id);
  };

  handleInputChange = e => {
    this.setState({
      todo: e.currentTarget.value
    });
  };

  render() {
    return (
      <div>
        <h1>TODOS</h1>
        <input
          type="text"
          onChange={this.handleInputChange}
          value={this.state.todo}
        />
        <button onClick={this.addTodo}>ADD</button>
        <ul>
          {this.props.todos.map(({ text }, id)=> (
            <li key={id} onClick={() => this.removeTodo(id)}>
              {text}
            </li>
          ))}
        </ul>
      </div>
    );
  }
}

const mapStateToProps = ({ todoList }) => ({
  todos: todoList.todos
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      addTodo,
      removeTodo
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(TodoList);
