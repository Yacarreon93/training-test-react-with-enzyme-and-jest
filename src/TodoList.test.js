import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { TodoList } from './TodoList';

// addTodo with button click
// removeTodo with li click
// matches snapshot

configure({ adapter: new Adapter() });

describe('<TodoList />', () => {
    it('calls addTodo action creator on button click', () => {
        const props = {
            addTodo: jest.fn(),
            todos: [],
        };

        const wrapper = shallow(<TodoList {...props} />);

        wrapper.find('input').simulate('change', { currentTarget: { value: 'buy groceries' }});
        wrapper.find('button').simulate('click');

        expect(props.addTodo).toHaveBeenCalledWith({ text: 'buy groceries' });
    });
    
    it('calls removeTodo action creator on li click', () => {
        const props = {
            removeTodo: jest.fn(),
            todos: [{ text: 'todo 1' }, { text: 'todo 2' }],
        };

        const wrapper = shallow(<TodoList {...props} />);
        
        wrapper.find('li').at(0).simulate('click');
        
        expect(props.removeTodo).toHaveBeenCalledWith(0);
    });
    
    it('matches snapshot', () => {
        const props = {
            todos: [],
        };

        const wrapper = shallow(<TodoList {...props} />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});