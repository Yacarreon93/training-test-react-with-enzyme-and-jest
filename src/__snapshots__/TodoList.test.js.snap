// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> matches snapshot 1`] = `
<div>
  <h1>
    TODOS
  </h1>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    ADD
  </button>
  <ul />
</div>
`;
